x-pack/legacy/plugins/apm/public/components/app/Settings/CustomizeUI/CustomActionsOverview/CustomActionsFlyout/FiltersSection.tsx

```diff
@@ -0,0 +1,220 @@
+  filters = [{ key: '', value: '' }],
+  onChange
+}: {
+  filters: Filter[];
+  onChange?: (filters: Filter[]) => void;
+}) => {
+  const handleAddFilter = () => {
+    if (typeof onChange === 'function') {
+      onChange([...filters, { key: '', value: '' }]);
+    }
+  };
+
+  return (
+    <>
+      <EuiTitle size="xs">
+        <h3>
+          {i18n.translate(
+            'xpack.apm.settings.customizeUI.customActions.flyout.filters.title',
+            {
+              defaultMessage: 'Filters'
+            }
+          )}
+        </h3>
+      </EuiTitle>
+      <EuiSpacer size="s" />
+      <EuiText size="xs">
+        {i18n.translate(
+          'xpack.apm.settings.customizeUI.customActions.flyout.filters.subtitle',
+          {
+            defaultMessage:
+              'Add additional values within the same field by comma separating values.'
+          }
+        )}
+      </EuiText>
+
+      <Filters filters={filters} onChange={onChange} />
+
+      <EuiSpacer size="xs" />
+
+      <AddFilterButton
+        onClick={handleAddFilter}
+        // Disable button when user has already added all items available
+        isDisabled={filters.length === filterOptions.length - 1}
+      />
+    </>
+  );
+};
+
+const AddFilterButton = ({
+  onClick,
+  isDisabled
+}: {
+  onClick: () => void;
+  isDisabled: boolean;
+}) => (
+  <EuiButtonEmpty
+    iconType="plusInCircle"
+    onClick={onClick}
+    disabled={isDisabled}
+  >
+    {i18n.translate(
+      'xpack.apm.settings.customizeUI.customActions.flyout.filters.addAnotherFilter',
+      {
+        defaultMessage: 'Add another filter'
+      }
+    )}
+  </EuiButtonEmpty>
+);
+
+const getSelectOptions = (filters: Filter[], idx: number) => {
+  return filterOptions.filter(option => {
+    const indexUsedFilter = filters.findIndex(
+      _filter => _filter.key === option.value
+    );
+    // Filter out all items already added, besides the one selected in the current filter.
+    return indexUsedFilter === -1 || idx === indexUsedFilter;
+  });
+};
+
+const Filters = ({
+  filters,
+  onChange
+}: {
+  filters: Filter[];
+  onChange?: (filters: Filter[]) => void;
+}) => {
+  const filterValueRefs = useRef<HTMLInputElement[]>([]);
+
+  const onChangeFilter = (filter: Filter, idx: number) => {
+    if (filterValueRefs.current[idx]) {
+      filterValueRefs.current[idx].focus();
+    }
+    const copyOfFilters = [...filters];
+    copyOfFilters[idx] = filter;
+    if (typeof onChange === 'function') {
+      onChange(copyOfFilters);
+    }
+  };
+
+  const onRemoveFilter = (idx: number) => {
+    const copyOfFilters = [...filters];
+    copyOfFilters.splice(idx, 1);
+    // When empty, means that it was the last filter that got removed,
+    // so instead of showing an empty list, will add a new empty filter.
+    if (isEmpty(copyOfFilters)) {
+      copyOfFilters.push({ key: '', value: '' });
+    }
+    if (typeof onChange === 'function') {
+      onChange(copyOfFilters);
+    }
+  };
```

Comment:
`onChangeFilter` and `onRemoveFilter` look overly complicated. Either we are doing something too custom (in which case we should not do it at this stage) or there is a simpler way to accomplish it.